povm_toolbox/library/randomized_projective_measurements.py

```diff
@@ -358,10 +374,13 @@ def _get_pvm_bindings_array(self, pvm_idx: np.ndarray) -> BindingsArray:
         theta: np.ndarray = np.empty(pvm_idx.shape)
         phi: np.ndarray = np.empty(pvm_idx.shape)
         for multi_index in np.ndindex(pvm_idx.shape):
-            # multi_index.shape = (*pv, num_batches, n_qubit)
+            # multi_index.shape is (*pv, num_batches, n_qubit)
             i_qubit = multi_index[-1]
-            theta[multi_index] = self.angles[i_qubit, pvm_idx[multi_index], 0]
-            phi[multi_index] = self.angles[i_qubit, pvm_idx[multi_index], 1]
+            theta[multi_index] = self.angles[i_qubit, pvm_idx[multi_index] % self._n_PVMs, 0]
+            # If the PVM is twirled, we flip the measurement:
+            if pvm_idx[multi_index] // self._n_PVMs > 0:
+                theta[multi_index] += np.pi
+            phi[multi_index] = self.angles[i_qubit, pvm_idx[multi_index] % self._n_PVMs, 1]
```

Comment:
I think I might prefer this a bit more explicitly:
```suggestion
            actual_pvm_idx = pvm_idx[multi_index] % self._n_PVMs
            twirl = pvm_idx[multi_index] // self._n_PVMs
            
            theta[multi_index] = self.angles[i_qubit, actual_pvm_idx, 0] + np.pi * (twirl > 0)
            phi[multi_index] = self.angles[i_qubit, actual_pvm_idx, 1]
```

This emphasizes the computation of `% n_PVMs` and `// n_PVMs` as shown in the dev comment earlier in the code. It also highlights the parallel between the indexing of `theta` and `phi`

Reply:
Looks cleaner I'll make the change 👍